### PR TITLE
Scottx611x/td api fixes

### PR DESCRIPTION
**Pull request by scottx611x**
Small fixes for ToolDefinition API code brought to light by writing documentation for it.

### CODE REVIEW

File: refinery/tool_manager/utils.py [hunk 1]
```diff
@@ -282,11 +282,13 @@ def get_workflow_list():
     return workflow_list
 
 
-def mock_get_workflow_list():
+def mock_get_workflow_list(filename):
```
Comment (hackdna) [inline refinery/tool_manager/utils.py]: This function simply reads a JSON file content and loads it into a Python object, so I'd name it accordingly and make it re-useable by allowing to specify any file path.

File: refinery/tool_manager/tests.py [hunk 2]
```diff
@@ -413,13 +413,26 @@ def test_invalid_workflow_step_annotation_c(self):
             )
 
     def test_generate_tool_definitions_management_command(self):
-        with mock.patch('tool_manager.utils.get_workflow_list',
-                        side_effect=mock_get_workflow_list):
+        with mock.patch(
+            'tool_manager.utils.get_workflow_list',
+            side_effect=[
```
Comment (scottx611x) [inline refinery/tool_manager/tests.py]: @hackdna: I could also  just nix the `mock_get_workflow_list()` and do the `json.loads()` in this `side_effects` array? thoughts?

Comment (hackdna) [inline refinery/tool_manager/tests.py]: Good point, that's probably even better in this case.

File: refinery/tool_manager/tests.py [hunk 3]
```diff
@@ -413,13 +412,57 @@ def test_invalid_workflow_step_annotation_c(self):
             )
 
     def test_generate_tool_definitions_management_command(self):
-        with mock.patch('tool_manager.utils.get_workflow_list',
-                        side_effect=mock_get_workflow_list):
+        invalid_workflows = open(
+            "tool_manager/test_data/invalid_galaxy_workflows.json"
```
Comment (hackdna) [inline refinery/tool_manager/tests.py]: DRY: `tool_manager/test_data/` should be stored in a variable.

**Pull request merged**